flow/util/template_filters.py

```diff
@@ -109,6 +109,61 @@ def calc_tasks(operations, name, parallel=False, allow_mixed=False):
         )
 
 
+def _parse_memory(memory):
+    """Parse memory from the memory flag passed by a user.
+
+    A valid memory flag passed to `FlowProject.submit` should have a suffix of either
+    "g" for gigabytes, or "m" for megabytes.
+
+    Parameters
+    ----------
+    memory : str
+        Required memory to reserve per node.
+
+    Returns
+    -------
+    float
+        The parsed memory (numeric value) in gigabytes.
+    """
+    try:
+        size_type = memory[-1]
+        if size_type.lower() == "m":
+            return float(memory[0:-1]) / 1024
+        elif size_type.lower() == "g":
+            return float(memory[0:-1])
+        else:
+            raise ValueError
+    except ValueError:
+        raise ValueError(
+            'Invalid memory passed. For gigabytes use suffix "g", '
+            'for megabytes use suffix "m".'
+        )
+    except TypeError:
+        raise
+
+
+def _calc_memory(operations, memory=None):
```

Comment:
This needs to be public API.
```suggestion
def calc_memory(operations, memory=None):
```

Reply:
Sounds good!